decisions/DR007-Hierarchical-or-compositional-traits-for-specifications.md

```diff
@@ -0,0 +1,361 @@
+# DR007 Hierarchical or composition based traits for specifications
+
+- **Status:** Proposed
+- **Impact:** High
+- **Driver:** @carmenpinto
+- **Approver:** @feltech
+- **Contributors:** @foundrytom @feltech @fnRaihanKibria
+- **Outcome:** A choice will be made for the specifications to be hierarchical or composition based
```

Comment:
Added summary of outcome in e7717b77781fa9927a5dd0e23863ca706d113557

Reply:
Ahh, sorry, It's meant to be a summary of what the change will be, so you don't need to read on. So something like

`OpenAssetIO will use a trait-composition based typing system instead of classical hierarchy.`

Reply:
Used this wording here: 70bea88